src/libcore/ptr/mod.rs

```diff
@@ -647,42 +647,47 @@ pub unsafe fn read<T>(src: *const T) -> T {
 /// [read-ownership]: ./fn.read.html#ownership-of-the-returned-value
 /// [valid]: ../ptr/index.html#safety
 ///
-/// # Examples
+/// ## On `packed` structs
 ///
-/// Access members of a packed struct by reference:
+/// Attempting to create a raw pointer to an `unaligned` struct field with
+/// an expression such as `&packed.unaligned as *const FieldType` creates an
+/// intermediate unaligned reference before converting that to a raw pointer.
+/// That this reference is temporary and immediately cast is inconsequential
+/// as the compiler always expects references to be properly aligned.
+/// As a result, using `&packed.unaligned as *const FieldType` causes immediate
+/// *undefined behavior* in your program.
 ///
-/// ```
-/// use std::ptr;
+/// An example of what not to do and how this relates to `read_unaligned` is:
```

Comment:
Users might wonder, after reading this: so how *can* I create a pointer to a field of a packed struct? The answer is "that's not possible"; I think that should be said explicitly somewhere.

Reply:
Sure, I'll add an explicit note about it.

(I was left a bit unsure by the weasel word "essentially" in your RFC. =P)

Reply:
The only "loophole" I can think of is, if you know the field offset (e.g. in case of a `#[repr(C, packed)]`), you can do manual raw pointer arithmetic to get a raw pointer to the field.